### PR TITLE
Integration with tpm2-tss

### PR DESCRIPTION
**Pull request by justsmth**
DRAFT: Integration tests will not pass until #1338 has been merged.

### Issues:
N/A

### Description of changes: 
* Add integration test for [tpm2-tss](https://github.com/tpm2-software/tpm2-tss).

### Call-outs:
N/A

### Testing:
* Ran tests on ubuntu host (rebased on #1338)
```
============================================================================
Testsuite summary for tpm2-tss 4.0.1-96-g166a5538-dirty
============================================================================
# TOTAL: 57
# PASS:  57
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/integration/tpm2_tss_patch/aws-lc-tpm2-tss.patch [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+diff --git a/src/tss2-esys/esys_crypto_ossl.c b/src/tss2-esys/esys_crypto_ossl.c
+index 1620788c..2924638a 100644
+--- a/src/tss2-esys/esys_crypto_ossl.c
++++ b/src/tss2-esys/esys_crypto_ossl.c
+@@ -8,6 +8,8 @@
+ #include <config.h>
+ #endif
+ 
++#include <openssl/mem.h>
++#include <openssl/bn.h>
```
Comment (andrewhop) [inline tests/ci/integration/tpm2_tss_patch/aws-lc-tpm2-tss.patch:22]: What was missing/in the wrong header without this?

File: tests/ci/integration/run_tpm2_tss_integration.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  cmake -DCMAKE_DEBUG_POSTFIX='' -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="${AWS_LC_INSTALL_FOLDER}" -DCMAKE_INSTALL_PREFIX="${CURL_INSTALL_FOLDER}" -B "${CURL_BUILD_FOLDER}" -S "${CURL_SRC_FOLDER}"
+  cmake --build "${CURL_BUILD_FOLDER}" --target install -j "${NUM_CPU_THREADS}"
+  ldd "${CURL_INSTALL_FOLDER}/lib/libcurl.so" | grep "${AWS_LC_INSTALL_FOLDER}/lib/libcrypto.so" || exit 1
+}
+
+function tpm2_tss_build() {
+  git apply "${SCRIPT_DIR}/tpm2_tss_patch/aws-lc-tpm2-tss.patch"
+  export PKG_CONFIG_PATH="${AWS_LC_INSTALL_FOLDER}/lib/pkgconfig:${CURL_INSTALL_FOLDER}/lib/pkgconfig"
+  /bin/sh ./bootstrap
+  ./configure --enable-unit --with-crypto=ossl CFLAGS="-g -ggdb -O0 -I\"${AWS_LC_INSTALL_FOLDER}\"/include -I\"${CURL_INSTALL_FOLDER}\"/include -L\"${AWS_LC_INSTALL_FOLDER}\"/lib -L\"${CURL_INSTALL_FOLDER}\"/lib" LT_SYS_LIBRARY_PATH="${LD_LIBRARY_PATH}"
```
Comment (andrewhop) [inline tests/ci/integration/run_tpm2_tss_integration.sh:48]: Why do you need the CFLAGS variable if you set the PKG_CONFIG_PATH? Is there an issue with our package config file that doesn't set everything tpm-tss needs?

File: tests/ci/integration/tpm2_tss_patch/aws-lc-tpm2-tss.patch (hunk 1 above)
Comment (andrewhop) [reply tests/ci/integration/tpm2_tss_patch/aws-lc-tpm2-tss.patch:40]: Could other projects hit the same issue? Does it make sense to "fix" our headers like https://github.com/aws/aws-lc/blob/main/include/openssl/evp.h#L65

**Review by WillChilds-Klein — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

File: tests/ci/integration/run_tpm2_tss_integration.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+mkdir -p "${SCRATCH_FOLDER}"
+rm -rf "${SCRATCH_FOLDER:?}"/*
+
+pushd "${SCRATCH_FOLDER}"
+
+function curl_build() {
+  cmake -DCMAKE_DEBUG_POSTFIX='' -DCMAKE_BUILD_TYPE=Debug -DCMAKE_PREFIX_PATH="${AWS_LC_INSTALL_FOLDER}" -DCMAKE_INSTALL_PREFIX="${CURL_INSTALL_FOLDER}" -B "${CURL_BUILD_FOLDER}" -S "${CURL_SRC_FOLDER}"
+  cmake --build "${CURL_BUILD_FOLDER}" --target install -j "${NUM_CPU_THREADS}"
+  ldd "${CURL_INSTALL_FOLDER}/lib/libcurl.so" | grep "${AWS_LC_INSTALL_FOLDER}/lib/libcrypto.so" || exit 1
```
[the remark this answers is not in the document]
Comment (justsmth) [reply tests/ci/integration/run_tpm2_tss_integration.sh:41]: I think tpm2-tss only uses our libcrypto directly, and uses libcurl for networking. The latest commit adds a check that libcurl is linked correctly.

**Review by andrewhop — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**